### PR TITLE
Add deposit CLI command

### PR DESCRIPTION
**Pull request by mat-if**
## Summary
image

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: ironfish-cli/src/commands/deposit.ts [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+  client: IronfishRpcClient | null = null
+  api: WebApi | null = new WebApi()
+
+  static flags = {
+    ...RemoteFlags,
+    fee: Flags.integer({
+      char: 'f',
+      default: 1,
+      description: `the fee amount in ORE, minimum of 1. 1 ORE is equal to ${MINIMUM_IRON_AMOUNT} IRON`,
```
Comment (leanthebean) [inline ironfish-cli/src/commands/deposit.ts:34]: We changed this to minimum of .1

Comment (NullSoldier) [reply ironfish-cli/src/commands/deposit.ts:34]: You are talking about the minimum transaction amount, this is the transaction fee.

File: ironfish-cli/src/commands/deposit.ts [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    const { canSend, errorReason } = await this.verifyCanSend(flags)
+    if (!canSend) {
+      Assert.isNotNull(errorReason)
+      this.log(errorReason)
+      this.exit(1)
+    }
+
+    const graffiti = this.sdk.config.get('blockGraffiti')
+
+    const balanceResp = await this.client.getAccountBalance()
```
Comment (NullSoldier) [inline ironfish-cli/src/commands/deposit.ts:77]: this needs to take the account name

File: ironfish-cli/src/commands/deposit.ts [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+      default: 1,
+      description: `the fee amount in ORE, minimum of 1. 1 ORE is equal to ${MINIMUM_IRON_AMOUNT} IRON`,
+    }),
+    expirationSequence: Flags.integer({
+      char: 'e',
+      description: 'max number of blocks for the transaction to wait before expiring',
+    }),
+    account: Flags.string({
+      char: 'a',
+      default: 'default',
```
Comment (NullSoldier) [inline ironfish-cli/src/commands/deposit.ts:42]: this needs to not have a default for most commands to pick up the default account.

File: ironfish-cli/src/commands/deposit.ts [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    }
+
+    const { canSend, errorReason } = await this.verifyCanSend(flags)
+    if (!canSend) {
+      Assert.isNotNull(errorReason)
+      this.log(errorReason)
+      this.exit(1)
+    }
+
+    const graffiti = this.sdk.config.get('blockGraffiti')
```
Comment (NullSoldier) [inline ironfish-cli/src/commands/deposit.ts:75]: Should this come from the RPC layer so it'll match the node that sends the transaction, or no?

File: ironfish-cli/src/commands/deposit.ts (hunk 3 above)
Comment (mat-if) [reply ironfish-cli/src/commands/deposit.ts]: Opted for loading the default account name from the RPC and passing it where needed to keep the validation logic simple. Explicit over implicit and all that.

Comment (NullSoldier) [reply ironfish-cli/src/commands/deposit.ts]: that sounds good to me!

**Review by NullSoldier — APPROVED** [review]
We can make further changes after this merges in.

**Pull request merged**